go/libraries/doltcore/sqle/tables.go

```diff
@@ -1422,3 +1444,129 @@ func (t *AlterableDoltTable) updateFromRoot(ctx *sql.Context, root *doltdb.RootV
 	t.WritableDoltTable.DoltTable = updatedTable.WritableDoltTable.DoltTable
 	return nil
 }
+
+func (t *AlterableDoltTable) CreateCheck(ctx *sql.Context, check *sql.CheckDefinition) error {
+	sch, err := t.table.GetSchema(ctx)
+	if err != nil {
+		return err
+	}
+
+	check = &(*check)
+	if check.Name == "" {
+		check.Name, err = t.generateCheckName(ctx, check)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = sch.Checks().AddCheck(check.Name, check.CheckExpression, check.Enforced)
+	if err != nil {
+		return err
+	}
+
+	newTable, err := t.table.UpdateSchema(ctx, sch)
+	if err != nil {
+		return err
+	}
+
+	root, err := t.db.GetRoot(ctx)
+	if err != nil {
+		return err
+	}
+
+	newRoot, err := root.PutTable(ctx, t.name, newTable)
+	if err != nil {
+		return err
+	}
+
+	err = t.db.SetRoot(ctx, newRoot)
+	if err != nil {
+		return err
+	}
+
+	return t.updateFromRoot(ctx, newRoot)
+}
+
+func (t *AlterableDoltTable) DropCheck(ctx *sql.Context, chName string) error {
+	sch, err := t.table.GetSchema(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = sch.Checks().DropCheck(chName)
+	if err != nil {
+		return err
+	}
+
+	newTable, err := t.table.UpdateSchema(ctx, sch)
+	if err != nil {
+		return err
+	}
+
+	root, err := t.db.GetRoot(ctx)
+	if err != nil {
+		return err
+	}
+
+	newRoot, err := root.PutTable(ctx, t.name, newTable)
+	if err != nil {
+		return err
+	}
+
+	err = t.db.SetRoot(ctx, newRoot)
+	if err != nil {
+		return err
+	}
+
+	return t.updateFromRoot(ctx, newRoot)
+}
+
+func (t *AlterableDoltTable) generateCheckName(ctx *sql.Context, check *sql.CheckDefinition) (string, error) {
+	var bb bytes.Buffer
+	bb.Write([]byte(check.CheckExpression))
+	hash := hash.Of(bb.Bytes())
+
+	hashedName := fmt.Sprintf("chk_%s", hash.String()[:8])
+	name := hashedName
+
+	var i int
+	for {
```

Comment:
You're not incrementing `i` in this loop

Reply:
Oof thanks bro